### PR TITLE
apt-dater: init at 1.0.3

### PR DESCRIPTION
**Pull request by enko**
###### Motivation for this change

I'm personaly using this nix package all the time and dont want it to get lost if my machine might breaks and I thought others might need this too.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/package-management/apt-dater/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  '';
+
+  postPatch = ''
+    substituteInPlace configure.ac --replace "/usr/bin/screen" "${screen}/bin/screen"
+  '';
+
+  doCheck = true;
+
+  meta = with stdenv.lib; {
+    homepage = https://www.ibh.de/apt-dater/;
```
Comment (rycee) [inline pkgs/tools/package-management/apt-dater/default.nix]: I get a 404 for this URL.

File: pkgs/tools/package-management/apt-dater/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    description = "provides an ncurses frontend for managing package updates on a large number of remote hosts using SSH";
```
Comment (rycee) [inline pkgs/tools/package-management/apt-dater/default.nix]: IMHO this is a bit long for a description. I'd suggest something like the description on their [github](https://github.com/DE-IBH/apt-dater): "Terminal-based remote package update manager". You can keep this description inside `longDescription`.

File: pkgs/tools/package-management/apt-dater/default.nix (hunk 1 above)
Comment (enko) [reply pkgs/tools/package-management/apt-dater/default.nix]: That is actually on ongoing issue of this project: https://github.com/DE-IBH/apt-dater/issues/123

File: pkgs/tools/package-management/apt-dater/default.nix (hunk 2 above)
Comment (enko) [reply pkgs/tools/package-management/apt-dater/default.nix]: I updated the descriptions from https://web.archive.org/web/20160811063553/http://www.ibh.de:80/apt-dater/ (the old homepage)

File: pkgs/tools/package-management/apt-dater/default.nix (hunk 1 above)
Comment (enko) [reply pkgs/tools/package-management/apt-dater/default.nix]: I will change the url to the github repo for now, till the issue is fixed.

**Comment by Mic92** [comment]
@GrahamcOfBorg build apt-dater

**Comment by Mic92** [comment]
@GrahamcOfBorg build apt-dater

**Comment by Mic92** [comment]
@enko can you please limit the build with `meta.platforms` to linux unless you indent to fix the build error on `darwin`?

**Comment by enko** [comment]
@Mic92 can I see the full output of the build on darwin, so I might see what causes the build failure?

File: pkgs/tools/package-management/apt-dater/default.nix [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+, automake115x, autoconf, pkgconfig, gettext
+, vim, glib, libxml2, openssl, ncurses, popt, screen
+}:
+
+stdenv.mkDerivation rec {
+  name = "apt-dater-${version}";
+  version = "1.0.3";
+
+  src = fetchurl {
+    url = "https://github.com/DE-IBH/apt-dater/archive/v${version}.tar.gz";
```
Comment (orivej) [inline pkgs/tools/package-management/apt-dater/default.nix]: github archives change over time, use fetchFromGitHub or fetchzip.

**Comment by Mic92** [comment]
@grahamc there is some log for the evaluation, but is the build output also fully logged?

File: pkgs/tools/package-management/apt-dater/default.nix [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+}:
+
+stdenv.mkDerivation rec {
+  name = "apt-dater-${version}";
+  version = "1.0.3";
+
+  src = fetchFromGitHub {
+    owner = "DE-IBH";
+    repo = "apt-dater";
+    rev = "v1.0.3";
```
Comment (peterhoeg) [inline pkgs/tools/package-management/apt-dater/default.nix]: Please change this to ```rev = "v${version}";```

**Review by matthewbauer — CHANGES_REQUESTED** [review]
merge conflict/see @peteroeg's comment

**Comment by c0bw3b** [comment]
@GrahamcOfBorg build apt-dater

**Comment by c0bw3b** [comment]
@GrahamcOfBorg build apt-dater

**Pull request merged**